models/Translation/Dao.php

```diff
@@ -173,8 +185,32 @@ public function createOrUpdateTable()
                           `text` text DEFAULT NULL,
                           `creationDate` int(11) unsigned DEFAULT NULL,
                           `modificationDate` int(11) unsigned DEFAULT NULL,
+                          `userOwner` int(11) unsigned DEFAULT NULL,
+                          `userModification` int(11) unsigned DEFAULT NULL,
                           PRIMARY KEY (`key`,`language`),
                           KEY `language` (`language`)
                         ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;");
     }
+
+    protected function updateModificationInfos()
+    {
+        $updateTime = time();
+        $this->model->setModificationDate($updateTime);
+
+        if (!$this->model->getCreationDate()) {
+            $this->model->setCreationDate($updateTime);
+        }
+
+        // auto assign user if possible, if no user present, use ID=0 which represents the "system" user
+        $userId = 0;
+        $user = \Pimcore\Tool\Admin::getCurrentUser();
```

Comment:
Alternatively you could maybe do something like:
`$userId = \Pimcore\Tool\Admin::getCurrentUser()?->getId() ?? 0`